webui/src/lib/components/repository/mergeResults.test.ts

```diff
@@ -0,0 +1,413 @@
+import { describe, it, expect } from 'vitest';
+import { mergeResults } from './mergeResults';
+import { Entry, ChangesData, EntryWithDiff } from './types';
+
+describe('mergeResults', () => {
+    describe('basic behavior', () => {
+        it('returns null when results is null', () => {
+            expect(mergeResults(null, { results: [] }, false)).toEqual([]);
+        });
+
+        it('returns undefined when results is undefined', () => {
```

Comment:
Results is not undefined here.  I think "results" refers to two things, so perhaps use a different name.